### PR TITLE
change(web): locked flicks 🐵

### PR DESCRIPTION
**Pull request by jahorton**
Per a recent team meeting to examine how flicks felt, etc, this PR reworks flicks from the form seen in #9909 and #9825 to something that will hopefully feel smoother and more intuitive.  To wit:

locked-flicks

There may be a bit of fine-tuning to be done (that first reset attempt on the 'i' seems a bit odd), but the following criteria have been met:

1. Flicks now "lock" into a specific direction.
    - Among other things, this allows us to make a nice, smoothed scrolling animation showing how close the user is to activating the flick.
2. This "lock" can be "reset" with a return to the base position.
    - A different flick direction may then be chosen while in the 'reset' state.
    - This has been updated since the recording above; flick-resets are now much cleaner and more reliable.
3. Attempting a flick in an unsupported direction, with no neighboring supported flick to serve as a reasonable alias, results in cancellation.
    - This occurs when the flick attempts to 'lock' its direction.
    - This gets dropped in #9972, at which point it simply "locks" on the base key until a valid flick direction is input.
4. There is a "fudge factor" around flicks, allowing their activation "reasonably near" to their correct angle.
    - This "fudge factor" also applies for locked-in flicks; the animation (and activation condition) thresholds are based on how much of the actual motion aligns with the locked-in direction.
        - This is done using the principles of [vector projection](https://en.wikipedia.org/wiki/Vector_projection).

There is a "fun" interaction between points 2 and 4 above:

locked-flick-reset-cancel

As the flick-reset occurs (due to vector projection) while quite far from the original, base position, and there is no flick available in the new direction, the flick quickly cancels upon reset.  That said, if there _were_ a flick available in the new direction (via temp-edit)...

jumpy-locked-flick-reset

It feels a _little_ abrupt, but not too much so.  It's probably "fine"... or at least, low-priority enough that further polish work is "safe" to triage until a later point in time.

----

Note: I've revised things since the screen capture above, but that was done alongside work on #9972 and is hard to fully display without that.  I've gone ahead and rebased the changes originally within #9991 onto this branch and done a `git merge -ff-only`; its code changes were near-entirely within the same files already changed here, so the review process should be simpler as a result... even if the child PR (#9972) is needed to fully appreciate certain aspects.

As such...

@keymanapp-test-bot skip

User tests will be specified either on #9972 or on its followup #9973.

### CODE REVIEW

**Comment by jahorton** [comment]
#9991's changeset has now been folded into this PR; it was a pretty easy rebase -> ff-only merge.

File: web/src/engine/osk/src/input/gestures/specsForLayout.ts [hunk 1]
```diff
@@ -271,17 +282,80 @@ export function flickStartContactModel(params: GestureParams): ContactModel {
...
+  let bestLockedDist = 0;
+
+  for(let i = 0; i < supportedDirs.length; i++) {
+    const dir = supportedDirs[i];
+    const lockedDist = calcLockedDistance(pathStats, dir);
+    if(lockedDist > bestLockedDist) {
+      bestLockedDist = lockedDist;
+      bestDir = dir;
+    }
+  }
```
Comment (mcdurdin) [inline web/src/engine/osk/src/input/gestures/specsForLayout.ts:303]:
```suggestion
  for(const dir of supportedDirs) {
    const lockedDist = calcLockedDistance(pathStats, dir);
    if(lockedDist > bestLockedDist) {
      bestLockedDist = lockedDist;
      bestDir = dir;
    }
  }
```

File: web/src/engine/osk/src/visualKeyboard.ts [hunk 2]
```diff
@@ -1245,6 +1236,7 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
 
     this.gestureParams.longpress.flickDist = 0.25 * this.currentLayer.rowHeight;
     this.gestureParams.flick.startDist     = 0.1  * this.currentLayer.rowHeight;
+    this.gestureParams.flick.dirLockDist   = 0.25 * this.currentLayer.rowHeight;
     this.gestureParams.flick.triggerDist   = 0.75 * this.currentLayer.rowHeight;
```
Comment (mcdurdin) [inline web/src/engine/osk/src/visualKeyboard.ts:1240]: Should these multipliers be consts at the top of the file?

**Review by mcdurdin — APPROVED** [review]
LGTM

**Pull request merged**